### ISSUE TITLE
1.8.11 doesn't allow access to package protected methods in java classes

### ISSUE THREAD

**Issue by zurbergram** [raphw/byte-buddy#463]
Recently bumped my version to 1.8.11 from 1.7.9 and had several Spock tests failing.  The tests failing where either stubbing the package protected method or calling it.

Below is the error message I received in the test that calls the package protected method `buildCakesFromCakeEntityList` in the `when:` block.

`tried to access method com.example.service.impl.CakeServiceImpl.buildCakesFromCakeEntityList(Ljava/util/List;)Lcom/example/protobuf/CakeService/Cakes; from class com.example.service.impl.CakeServiceImpl$SpockMock$dXVHAYpP
java.lang.IllegalAccessError: tried to access method com.example.service.impl.CakeServiceImpl.buildCakesFromCakeEntityList(Ljava/util/List;)Lcom/example/protobuf/CakeService/Cakes; from class com.example.service.impl.CakeServiceImpl$SpockMock$dXVHAYpP
    at org.spockframework.mock.runtime.ByteBuddyMethodInvoker.respond(ByteBuddyMethodInvoker.java:29)
    at org.spockframework.mock.runtime.MockInvocation.callRealMethod(MockInvocation.java:60)
    at org.spockframework.mock.CallRealMethodResponse.respond(CallRealMethodResponse.java:29)
    at org.spockframework.mock.runtime.MockController.handle(MockController.java:49)
    at org.spockframework.mock.runtime.JavaMockInterceptor.intercept(JavaMockInterceptor.java:74)
    at org.spockframework.mock.runtime.ByteBuddyInterceptorAdapter.interceptNonAbstract(ByteBuddyInterceptorAdapter.java:35)
    at com.example.service.impl.CakeServiceImplSpec.buildCakesFromCakeEntityList(CakeServiceImplSpec.groovy:256)`

**Comment by raphw** [comment]
I had to change the default class loading strategy to keep the standard API free from use of the `sun.misc.Unsafe` class which was partially removed in one of the recent Java 11 EAs. Spock needs to update to the method handle API as I described it in: https://mydailyjava.blogspot.no/2018/04/jdk-11-and-proxies-in-world-past.html - Unfortunately, this cannot be done in Byte Buddy but must be applied by Spock due to the call site sensitivity of method handle lookups.

I will send them a PR. Thanks for reporting.
